webui/react/src/e2e/models/BaseComponent.ts

```diff
@@ -38,7 +38,7 @@ export type NamedComponentArgs =
  * @param {CanBeParent} obj.parent - The parent used to locate this BaseComponent
  * @param {string} obj.selector - Used as a selector uesd to locate this object
  */
-export class BaseComponent implements ModelBasics {
+export class BaseComponent implements ComponentBasics {
```

Comment:
Not really actionable at the moment, and this is a very small example, but this is the beginning of the structure that becomes problematic with POM(and to some extent OOP in general). A large tree of inheritance with a bunch of intermediate classes is necessary, but it doesn't really encapsulate anything different conceptually, so it's difficult to know why it is this way and difficult to  figure out where different functions/properties come from in the tree. We should try not to insert more intermediate classes in this tree if we can.